react/features/filmstrip/components/web/Filmstrip.js

```diff
@@ -306,37 +433,112 @@ class Filmstrip extends PureComponent <Props> {
-                    id = 'remoteVideos'>
-                    {!_disableSelfView && (
+                {_resizableFilmstrip
+                    ? <div className = { classes.filmstripContainer }>
                         <div
-                            className = 'filmstrip__videos'
-                            id = 'filmstripLocalVideo'>
-                            <div id = 'filmstripLocalVideoThumbnail'>
-                                {
-                                    !tileViewActive && <Thumbnail
-                                        key = 'local' />
-                                }
-                            </div>
-                        </div>
-                    )}
-                    {
-                        this._renderRemoteParticipants()
-                    }
-                </div>
-                <AudioTracksContainer />
+                            className = { clsx('dragHandle',
+                                classes.dragHandle,
+                                isMouseDown && 'show')
+                            }
+                            onMouseDown = { this._onDragHandleMouseDown } />
+                        {filmstrip}
+                    </div>
+                    : filmstrip
+                }
             </div>
         );
     }
 
+    _onDragHandleMouseDown: (MouseEvent) => void;
+
+    /**
+     * Handles mouse down on the drag handle.
+     *
+     * @param {MouseEvent} e - The mouse down event.
+     * @returns {void}
+     */
+    _onDragHandleMouseDown(e) {
+        this._dragRef.current.down = true;
+        this._dragRef.current.initialPosition = e.clientX;
+        this._dragRef.current.initialWidth = this.props._verticalFilmstripWidth || DEFAULT_FILMSTRIP_WIDTH;
+        this.setState({
+            isMouseDown: true
+        });
+    }
+
+    _onDragMouseUp: () => void;
+
+    /**
+     * Drag handle mouse up handler.
+     *
+     * @returns {void}
+     */
+    _onDragMouseUp() {
+        if (this._dragRef.current.down) {
+            this._dragRef.current.down = false;
+            this.setState({
+                isMouseDown: false
+            });
+        }
+    }
+
+    _onFilmstripResize: (MouseEvent) => void;
+
+    /**
+     * Handles drag handle mouse move.
+     *
+     * @param {MouseEvent} e - The mousemove event.
+     * @returns {void}
+     */
+    _onFilmstripResize(e) {
+        if (this._dragRef.current.down) {
+            const { dispatch, _verticalFilmstripWidth, _maxFilmstripWidth } = this.props;
+            const diff = this._dragRef.current.initialPosition - e.clientX;
+            const width = Math.max(
+                Math.min(this._dragRef.current.initialWidth + diff, _maxFilmstripWidth),
+                DEFAULT_FILMSTRIP_WIDTH
+            );
+
+            if (width !== _verticalFilmstripWidth) {
+                dispatch(setUserFilmstripWidth(width));
+                VideoLayout.refreshLayout();
```

Comment:
Is there a way to move this in a middleware or something since it is legacy code?